pylint/checkers/variables.py

```diff
@@ -1526,21 +1503,20 @@ def _check_consumer(
                 node=node,
                 confidence=confidence,
             )
-            if current_consumer.consumed_uncertain[node.name]:
-                # If there are nodes added to consumed_uncertain by
-                # get_next_to_consume() because they might not have executed,
-                # return a CONSUME action so that _undefined_and_used_before_checker()
-                # will mark them as used
-                return (VariableVisitConsumerAction.CONSUME, found_nodes)
-            return (VariableVisitConsumerAction.RETURN, None)
+            # Mark for consumption any nodes added to consumed_uncertain by
+            # get_next_to_consume() because they might not have executed.
+            return (
+                VariableVisitConsumerAction.RETURN,
+                current_consumer.consumed_uncertain[node.name],
```

Comment:
Is this the same as `current_consumer.get_next_to_consume(node)`?

Reply:
I don't think so: that function has already run and actually was responsible for _populating_ the dict entry being accessed here. Now having received a RETURN action, it's time to finish consuming all the names, including what maybe-executed.

I'm realizing `consumed_uncertain` is less descriptive as a name now that we're getting rid of CONSUME as an action. It now refers more to "execution uncertain" or something.